src/oot/ovl/kaleido_scope.c

```diff
@@ -150,3 +150,1340 @@ static void KaleidoScope_HandleDungeonChests(GameState_Play* play)
+                sStickXRepeatState = -1;
+            }
+        } else if (pauseCtx->stickAdjX > 30) {
+            if (sStickXRepeatState == 1) {
+                sStickXRepeatTimer--;
+                if (sStickXRepeatTimer < 0) {
+                    // Allow the input to register and apply the delay for all subsequent repeated inputs
+                    sStickXRepeatTimer = 2;
+                } else {
+                    // Cancel the current input
+                    pauseCtx->stickAdjX = 0;
+                }
+            } else {
+                // Allow the input to register and apply the delay for the first repeated input
+                sStickXRepeatTimer = 10;
+                sStickXRepeatState = 1;
+            }
+        } else {
+            sStickXRepeatState = 0;
+        }
+
+        if (pauseCtx->stickAdjY < -30) {
+            if (sStickYRepeatState == -1) {
+                sStickYRepeatTimer--;
+                if (sStickYRepeatTimer < 0) {
+                    // Allow the input to register and apply the delay for all subsequent repeated inputs
+                    sStickYRepeatTimer = 2;
+                } else {
+                    // Cancel the current input
+                    pauseCtx->stickAdjY = 0;
+                }
+            } else {
+                // Allow the input to register and apply the delay for the first repeated input
+                sStickYRepeatTimer = 10;
+                sStickYRepeatState = -1;
+            }
+        } else if (pauseCtx->stickAdjY > 30) {
+            if (sStickYRepeatState == 1) {
+                sStickYRepeatTimer--;
+                if (sStickYRepeatTimer < 0) {
+                    // Allow the input to register and apply the delay for all subsequent repeated inputs
+                    sStickYRepeatTimer = 2;
+                } else {
+                    // Cancel the current input
+                    pauseCtx->stickAdjY = 0;
+                }
+            } else {
+                // Allow the input to register and apply the delay for the first repeated input
+                sStickYRepeatTimer = 10;
+                sStickYRepeatState = 1;
+            }
+        } else {
+            sStickYRepeatState = 0;
+        }
+    }
+
+    switch (pauseCtx->state)
+    {
+        case PAUSE_STATE_OWLWARP_2:
+            u8* sUnpausedButtonStatus = OverlayAddr(0x80830150);
+            sUnpausedButtonStatus[0] = gSaveContext.buttonStatus[0];
+            sUnpausedButtonStatus[1] = gSaveContext.buttonStatus[1];
+            sUnpausedButtonStatus[2] = gSaveContext.buttonStatus[2];
+            sUnpausedButtonStatus[3] = gSaveContext.buttonStatus[3];
+            sUnpausedButtonStatus[4] = gSaveContext.buttonStatus[4];
+
+            pauseCtx->map_x = 0;
+            pauseCtx->map_hilite = pauseCtx->map_cursor = pauseCtx->dungeonMapSlot = VREG(30) + 3;
+
+            WREG(16) = -175;
+            WREG(17) = 155;
+
+            u32 src = 0x7BD000;
```

Comment:
Would probably be cleaner to use kComboDmaData here too, even though it's a native file.